File: fight_env/player/player_model.py
from dataclasses import dataclass, field
from typing import Optional, List

from fight_env.player.refs.events import Event, Response
from fight_env.player.refs.intents import Intent
from fight_env.player.stats import Stats
from fight_env.player.refs.tasks import FighterTask, TaskTimeline


@dataclass
class PlayerModel:
    task: FighterTask = FighterTask.NONE
    timeline: TaskTimeline = field(default_factory=TaskTimeline)
    stats: Stats = field(default_factory=Stats)
    current_event: Event = field(default_factory=list)
    current_responses: List[Response] = field(default_factory=list)
    requested_action: Optional[Intent] = field(default=None)

    stamina_cost_frame: int = 0
    stamina_cost_enter_task: int = 0

    is_dead: bool = False

    hp: int = 0
    stamina: int = 0

File: unity/Assets/FightCore/PlayerModel.cs
// Port of fight_env/player/player_model.py and player_snapshot.py.
using System.Collections.Generic;

namespace FightCore
{
    public sealed class PlayerModel
    {
        public FighterTask Task = FighterTask.NONE;
        public TaskTimeline Timeline = new TaskTimeline();
        public Stats Stats;
        public Event CurrentEvent;
        public List<Response> CurrentResponses = new List<Response>();
        public Intent RequestedAction;

        public int StaminaCostFrame;
        public int StaminaCostEnterTask;

        public bool IsDead;

        public int Hp;
        public int Stamina;
    }

    // fight_env/player/player_snapshot.py :: PlayerSnapshot
    public sealed class PlayerSnapshot
    {
        public readonly int Hp;
        public readonly int MaxHp;
        public readonly int Stamina;
        public readonly int MaxStamina;
        public readonly FighterTask Task;
        public readonly int FrameOffset;
        public readonly string Name;

        public PlayerSnapshot(PlayerModel model)
        {
            Hp = model.Hp;
            MaxHp = model.Stats.MaxHp;
            Stamina = model.Stamina;
            MaxStamina = model.Stats.MaxStamina;
            Task = model.Task;
            Name = model.Stats.Name;
            FrameOffset = model.Timeline.FrameOffset;
        }
    }
}